javalib/src/main/scala/java/util/BitSet.scala

```diff
@@ -0,0 +1,694 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package java.util
+
+import java.io.Serializable
+import java.lang.Integer.bitCount
+import java.lang.Integer.toUnsignedLong
+import java.nio.{ByteBuffer, LongBuffer}
+import java.util
+import java.util.ScalaOps.IntScalaOps
+
+private object BitSet {
+  private final val AddressBitsPerWord = 5 // Int Based 2^5 = 32
+  private final val ElementSize = 1 << AddressBitsPerWord
+  private final val RightBits = ElementSize - 1
+
+  def valueOf(longs: Array[Long]): util.BitSet = {
+    val bs = new util.BitSet
+
+    for (i <- 0 until longs.length * 64) {
+      val idx = i / 64
+      if ((longs(idx) & (1L << (i % 64))) != 0)
+        bs.set(i)
+    }
+
+    bs
+  }
+
+  def valueOf(lb: LongBuffer): BitSet = {
+    if (lb.hasArray()) valueOf(lb.array())
+    else {
+      val arr = new Array[Long](lb.remaining())
+      lb.get(arr)
+      valueOf(arr)
+    }
+  }
+
+  def valueOf(bytes: Array[Byte]): BitSet = {
+    val bs = new BitSet
+
+    for (i <- 0 until bytes.length * 8) {
+      val idx = i / 8
+      if ((bytes(idx) & (1 << (i % 8))) != 0)
+        bs.set(i)
+    }
+
+    bs
+  }
+
+  def valueOf(bb: ByteBuffer): BitSet = {
+    if (bb.hasArray()) valueOf(bb.array())
+    else {
+      val arr = new Array[Byte](bb.remaining())
+      bb.get(arr)
+      valueOf(arr)
+    }
+  }
```

Comment:
This is suspicious: it will modify the position of the passed Buffer only if it is not backed by an Array. Is that intended?

Reply:
good catch, I forgot to add a bb.rewind(), let me double check the javadocs and I'll push an update

Reply:
Here's a quick and dirty scastie as a proof of concept for JVM runtime: https://scastie.scala-lang.org/3ronEAsySBu6IX8cgGdwCw

Reply:
@gzm0 updated with better tests, thanks!